ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/ft/FtLineModelListener.java

```diff
@@ -63,10 +64,45 @@ public void exitFt_key(FtLineParser.Ft_keyContext ctx) {
     public void exitFt_note(FtLineParser.Ft_noteContext ctx) {
         if (ctx.note_text() != null) {
             String result = ctx.note_text().getText();
+            result = replaceVarSeqSpaceNoSpace(result);
             result = result.replaceAll("''", "\"");
             ft.setFtText(updateAltSeqText(result));
         }
     }
+    
+    private String replaceVarSeqSpaceNoSpace(String text1) {
+        if(Strings.isNullOrEmpty(text1)){
+            return text1;
+        }
+        String newText = text1;
+        while(newText.contains(RememberLastTokenLexer.VAR_SEQ_SPACE_NO_SPACE_PLACEHOLDER)) {
+           String restText = newText.substring(newText.indexOf(RememberLastTokenLexer.VAR_SEQ_SPACE_NO_SPACE_PLACEHOLDER)
+                   +RememberLastTokenLexer.VAR_SEQ_SPACE_NO_SPACE_PLACEHOLDER.length() );
+           String token = restText;
+           int indexSpace = restText.indexOf(" ");
+           int indexLess = restText.indexOf("<");
+           if(indexSpace>0) {
+               if(indexLess>0) {
+                   if(indexLess<indexSpace) {
+                       token =restText.substring(0, restText.indexOf("<"));
+                   }else {
+                       token =restText.substring(0, restText.indexOf(" "));
+                   }
+               }else {
+                   token =restText.substring(0, restText.indexOf(" "));
+               }
+           } else  if(indexLess>0) {
+               token =restText.substring(0, restText.indexOf("<"));
+           }
+       
+          if(RememberLastTokenLexer.isSequenceLetter(token)) {
+              newText = newText.replace(RememberLastTokenLexer.VAR_SEQ_SPACE_NO_SPACE_PLACEHOLDER, "");
```

Comment:
Jie, I think here we should use replace first, right? because the replace method will replace all occurrences of VAR_SEQ_SPACE_NO_SPACE_PLACEHOLDER and the looking will not make sense. because there will never be a second iteration in the looping. 
Currently we do not have a test the covers iterating multiple times in the looping, can we add it?

Reply:
I added a multiline test case